### PR TITLE
[FLINK-23312][ci] speed up compilation for e2e tests

### PR DESCRIPTION
**Pull request by NicoK**
## What is the purpose of the change

The "e2e" builder in Azure pipelines builds Flink again on top of what the "compile" builder is already doing. This unnecessary duplicates a couple of checks that are enough to execute once and can be skipped via providing `-Dfast` (and `-Pskip-webui-build` as suggested by the review comments).

On my local machine with 32GB RAM, 8 physical cores and a fast NVMe SSD, the difference is pretty big:

```
time mvn clean install -Dscala-2.12 -DskipTests -pl flink-dist -am
# -> 6:40 min

time mvn clean install -Dscala-2.12 -DskipTests -Dfast -pl flink-dist -am
# -> 5:40 min
```

I looked at a couple of Azure Pipeline runs and it seems like this for the "Build Flink" task in the "e2e_ci" builder:
- without this PR: 29-33 min
- with `-Dfast`: 20-25 min
- with `-Dfast` and `-Pskip-webui-build`: 16-20min

## Brief change log

- use `-Dfast` and `-Pskip-webui-build` for the e2e build

## Verifying this change

This change does not affect Flink code and instead influences the Azure pipeline definition

### CODE REVIEW

**Comment by NicoK** [comment]
FYI: since this PR couldn't have introduced any test failure, I created https://issues.apache.org/jira/browse/FLINK-23331 for the one that Azure is reporting.

**Review by zentol — APPROVED** [review]
+1, but we could also skip building the UI with `-Pskip-webui-build`

**Comment by NicoK** [comment]
@zentol Is that not needed by any E2E test? Also, is that guaranteed for the future?

**Comment by zentol** [comment]
> Is that not needed by any E2E test? Also, if the answer is "yes": is that guaranteed for the future?

I think we neither have tests for the UI nor will need it for the future. It really just skips building of the UI; all parts of the REST API will continue to work.

**Comment by NicoK** [comment]
> I think we neither have tests for the UI nor will need it for the future. It really just skips building of the UI; all parts of the REST API will continue to work.

In that case, we could add it as well - probably in a follow-up PR?

**Comment by zentol** [comment]
> probably in a follow-up PR

nah I'd just add it in this PR and sell the whole thing as "speed up compilation for e2e tests"

**Comment by NicoK** [comment]
Thanks, I just added `-Pskip-webui-build` to this PR and will wait for another round of CI before I merge this and all the backport PRs.

**Comment by NicoK** [comment]
@zentol Now that I added `-Pskip-webui-build`, there are a few new test failures that seem related, e.g.
```
Jul 13 12:45:37 ================================================================================
Jul 13 12:45:37 Test testVCoresAreSetCorrectlyAndJobManagerHostnameAreShownInWebInterfaceAndDynamicPropertiesAndYarnApplicationNameAndTaskManagerSlots(org.apache.flink.yarn.YARNSessionCapacitySchedulerITCase) is running.
Jul 13 12:45:37 --------------------------------------------------------------------------------
Jul 13 12:45:44 12:45:44,543 [                main] INFO  org.apache.flink.yarn.YARNSessionCapacitySchedulerITCase     [] - Extracted hostname:port: 4fe9265b2ae6:34779

------------------------------------------------------------
 The program finished with the following exception:
Caused by: org.apache.flink.runtime.client.JobSubmissionException: Failed to submit JobGraph.
	at org.apache.flink.client.program.rest.RestClusterClient.lambda$submitJob$9(RestClusterClient.java:405)
	at java.util.concurrent.CompletableFuture.uniExceptionally(CompletableFuture.java:884)
	at java.util.concurrent.CompletableFuture$UniExceptionally.tryFire(CompletableFuture.java:866)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
	at org.apache.flink.util.concurrent.FutureUtils.lambda$retryOperationWithDelay$9(FutureUtils.java:385)
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:774)
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:750)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
	at org.apache.flink.runtime.rest.RestClient.lambda$submitRequest$1(RestClient.java:430)
	at org.apache.flink.shaded.netty4.io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:577)
	at org.apache.flink.shaded.netty4.io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:570)
	at org.apache.flink.shaded.netty4.io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:549)
	at org.apache.flink.shaded.netty4.io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:490)
	at org.apache.flink.shaded.netty4.io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:615)
	at org.apache.flink.shaded.netty4.io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:608)
	at org.apache.flink.shaded.netty4.io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:117)
	at org.apache.flink.shaded.netty4.io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.fulfillConnectPromise(AbstractNioChannel.java:321)
	at org.apache.flink.shaded.netty4.io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:337)
	at org.apache.flink.shaded.netty4.io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:702)
	at org.apache.flink.shaded.netty4.io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650)
	at org.apache.flink.shaded.netty4.io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576)
	at org.apache.flink.shaded.netty4.io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
	at org.apache.flink.shaded.netty4.io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at org.apache.flink.shaded.netty4.io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.flink.util.concurrent.FutureUtils$RetryException: Could not complete the operation. Number of retries has been exhausted.
	at org.apache.flink.util.concurrent.FutureUtils.lambda$retryOperationWithDelay$9(FutureUtils.java:381)
	... 21 more
Caused by: java.util.concurrent.CompletionException: org.apache.flink.shaded.netty4.io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:8081
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniCompose(CompletableFuture.java:957)
	at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:940)
	... 19 more
Caused by: org.apache.flink.shaded.netty4.io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:8081
Caused by: java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:716)
	at org.apache.flink.shaded.netty4.io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:330)
	at org.apache.flink.shaded.netty4.io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
	at org.apache.flink.shaded.netty4.io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:702)
	at org.apache.flink.shaded.netty4.io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650)
	at org.apache.flink.shaded.netty4.io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576)
	at org.apache.flink.shaded.netty4.io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
	at org.apache.flink.shaded.netty4.io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at org.apache.flink.shaded.netty4.io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at java.lang.Thread.run(Thread.java:748)
```
from https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=20373&view=logs&j=fc5181b0-e452-5c8f-68de-1097947f6483&t=62110053-334f-5295-a0ab-80dd7e2babbf&l=28585

**Comment by zentol** [comment]
It can't be related for the simple reason that the test failure did not occur in the e2e stage.

**Comment by NicoK** [comment]
ok - I'm lacking the insights to argue with you here ;)

In that case, I guess, we can merge this, right?

**Pull request merged**
